AppController/djinn.rb

```diff
@@ -1922,37 +1936,48 @@ def job_start(secret)
       # the state in case we are looking for a zookeeper server.
       pick_zookeeper(@zookeeper_data)
 
-      @state = "Done starting up AppScale, now in heartbeat mode"
-
       # We save the current @options and roles to check if
       # restore_appcontroller_state modifies them.
       old_options = @options.clone
       old_jobs = my_node.jobs
 
       # The following is the core of the duty cycle: start new apps,
       # restart apps, terminate non-responsive AppServers, and autoscale.
-      # Every other node syncs its state with the login node state.
+
+      # Every other node syncs its state with the login node state. The
+      # load_balancers needs to check the applications that got loaded
+      # this time, to setup the routining.
+      my_versions_loaded = @versions_loaded if my_node.is_load_balancer?
```

Comment:
I can't figure out why this would be needed. Isn't `@versions_loaded` only changed for a given appcontroller when that process actually sets up or stops the version?

Reply:
@version_loaded gets updated when the lb loads the status from zookeeper (ie shadow), so it will inherits what the master has done this duty-cycle.